sdk/storage/Azure.Storage.DataMovement/src/TransferJobInternal.cs

```diff
@@ -504,6 +504,7 @@ await TransferFailedEventHandler.RaiseAsync(
                         .ConfigureAwait(false);
                 }
             }
+            _dataTransfer.TransferStatus.SetFailedItem();
```

Comment:
Do we want to do this here? Since I believe that InvokeFailedArgAsync() also gets called during Pause when we cancel the internal CancellationToken and throw an exception, and I don't think we want to set Failed item in this scenario. 

Seems like we should move this up in the if-statement.

Reply:
Strange that no test caught this. I don't know enough to know exactly what to test. @amnguye do you have ideas on this? Happy to submit a separate PR that catches this.

Reply:
Just submit a separate PR to test this.